components/TextArea/TextArea.tsx

```diff
@@ -0,0 +1,47 @@
+import { useRef, useState } from "react";
+import styles from './TextArea.module.css'
+import { useEffect } from "react";
+
+export type TextAreaProps = {
+    maxLength: number;
+    text: string;
+}
+
+const autosizeTextArea = (
```

Comment:
This should be renamed to `useAutosizeTextArea` (or something else that starts with `use`) because it calls the `useEffect`